staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go

```diff
@@ -133,14 +133,22 @@ func (f *FieldManager) Update(liveObj, newObj runtime.Object, manager string) (r
 	}
 
 	// TODO(apelisse) use the first return value when unions are implemented
-	_, managed.Fields, err = f.updater.Update(liveObjTyped, newObjTyped, apiVersion, managed.Fields, manager)
+	_, managed.Fields, err = f.updater.Update(liveObjTyped, newObjTyped, apiVersion, managed.Fields, "self")
 	if err != nil {
 		return nil, fmt.Errorf("failed to update ManagedFields: %v", err)
 	}
-	managed.Fields = f.stripFields(managed.Fields, manager)
+	managed.Fields = f.stripFields(managed.Fields, "self")
 
-	// Update the time in the managedFieldsEntry for this operation
-	managed.Times[manager] = &metav1.Time{Time: time.Now().UTC()}
+	// Merge the managedFieldsEntry for this operation with the previous
+	// operations from the same manager if the object changed
+	if f, ok := managed.Fields["self"]; ok {
+		if old, ok := managed.Fields[manager]; ok {
+			f = fieldpath.NewVersionedSet(f.Set().Union(old.Set()), f.APIVersion(), f.Applied())
```

Comment:
Interesting how we could remove that union from smd now

Reply:
Yeah we could do that and simplify things, just need to make sure that people don't give the same manager name to smd as any existing manager for updates (maybe return an error?)

Reply:
I think it's fine. As long as we never send the same manager, the union is always going to be a no-op?